### PR TITLE
kqueue main loop done

### PR DESCRIPTION
**Pull request by Tatamize**
Hi, It is still messy but the kqueue main part is done.

Now, 2 server's dammy information is set, maybe the information for servers is not enough but please never mind...

after running the ./webserv , if you access to "http://localhost:8080" by firefox, the "hello" response will be sent to the browser.  

you can try to send many many access at once with TEST_client go file.

I changed Makefile, maybe I should put cpp files in src folder later...

### CODE REVIEW

File: .DS_Store
Comment (Zangetsu89) [inline .DS_Store:1]: pls remove this file.

File: include/SockListen.hpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	Server				*_num_Server;
+	int					_num_socket;
+	struct sockaddr_in	_listen_sockaddr;
+	socklen_t 			_listen_sockaddr_len;
+	struct kevent		_kev_listen;
+	
+
+	// base public member function
+	public:
+	SockListen();				// do not use : SockListen must be created with socket number
```
Comment (Zangetsu89) [inline include/SockListen.hpp:30]: Then we move the constructor to 'protected:' right?

File: include/Server.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# include <sys/event.h>
+# include <sys/time.h>
+# include <stdlib.h>
+# include <unistd.h>
+# include <netdb.h>
+# include <signal.h>
+# include "SockConnect.hpp"
+# include "SockListen.hpp"
+
+class SockConnect;
```
Comment (Zangetsu89) [inline include/Server.hpp:16]: I see SockConnect is forward declared even tho not used

File: include/Server.hpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# include <iostream>
+# include <sys/socket.h>
+# include <sys/types.h>
+# include <sys/event.h>
+# include <sys/time.h>
+# include <stdlib.h>
+# include <unistd.h>
+# include <netdb.h>
+# include <signal.h>
+# include "SockConnect.hpp"
```
Comment (Zangetsu89) [inline include/Server.hpp:13]: can you include header and do forward declaration?

File: include/.DS_Store
Comment (Zangetsu89) [inline include/.DS_Store:1]: remove please

File: src/.DS_Store
Comment (Zangetsu89) [inline src/.DS_Store:1]: please remove

**Review by Zangetsu89 — CHANGES_REQUESTED** [review]
I noticed that you didn't pull from main before you pushed this, so please pull from main and adjust accordingly.
Please create a feature branch, split this pull request into sub branches and push one class at a time, they should be individual classes not inheriting from each other yet, when all are merged to the feature branch, you can added the inheritance part. this will make it clear for us what the classes are supposed to do and work.
When you create a cpp file, please place it in the src folder, in a subfolder if necessary as well.

File: .DS_Store
Comment (Tatamize) [reply .DS_Store:1]: Thanks, I'm trying to push the files by each class. I will re-push these files without .DS_store.
Can I remove this pull request and branch?

File: include/SockListen.hpp (hunk 1 above)
Comment (Tatamize) [reply include/SockListen.hpp:30]: Thanks! I will modify it in the new file. (I will push it by new branch)

File: include/Server.hpp (hunk 2 above)
Comment (Tatamize) [reply include/Server.hpp:16]: Thanks! I will modify these double header inclusions in the new file. (I will push it by new branch)

File: include/Server.hpp (hunk 3 above)
Comment (Tatamize) [reply include/Server.hpp:13]: Yes as previous comment, I will modify these double header inclusions in the new file. (I will push it by new branch)

File: include/.DS_Store
Comment (Tatamize) [reply include/.DS_Store:1]: OK! (I will do it in new branch)

File: src/.DS_Store
Comment (Tatamize) [reply src/.DS_Store:1]: OK! sorry for several times (I will do it in new branch)

**Comment by Tatamize** [comment]
I made other pull requests so close this one

**Pull request closed without merging**